slides/src/content/ngrx.md

```diff
@@ -0,0 +1,161 @@
+# Redux (@ngrx)
+
+---
+
+## Motivation 
+
+1. Create [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) 
+1. [Separation of Concern](https://en.wikipedia.org/wiki/Separation_of_concerns) : Separation of Data and Logic
+1. Improve scalability by having things _react_ to events as opposed to caller dictating actions
+1. Reduce Complexity of Mental Map by promoting separation of responsibility between components
+1. Create a coherent framework so future developers will know how to add features by following convention
+1. [Uni-Directional Data Flow](http://redux.js.org/docs/basics/DataFlow.html) : Easy to reason about the interractions between actors
+
+---
+
+## Structure of Systems - Redux vs Server Architecture
+
+| Function | Angular With `@ngrx`  | Server |
+|---|---|---|
+| Data Storage | Store  |  Database  | 
+| Presentation View |  `@Component`  | Client API  |
+| Logic to massage data for Storage |  `@Effects`  | Services |
+| Communication | Action / Observables  | Function Invocation | 
+
+---
+
+## Redux as a System 
+
+<img src="content/images/redux-and-data-flow.png" width="50%"/>
+
+---
+
+## Setup Ngrx Dev-Tools
+
+FIXME
+
+---
+
+## Actions
+
+1. Responsible for Communication between parts of redux
+1. Implemented with [Observable](http://reactivex.io/rxjs/) streams
+1. Use actions instead of function Invocation
+1. Listeners are responsible to determine how to react to actions
+1. Has 2 properties, `type`, and `payload`
+    1. `type` is the identifier the listeners can use to listen to actions
+    1. `payload` is a flexible object that carries information
+
+---
+
+## Dispatching An Action
+
+```ts
+import { Store } from '@ngrx/store';
+import { people } from './people';
+
+@Component({
+    selector: 'person-input',
+    template: `
+      <button (click)="add('John')">Add Person</button>
+    `
+})
+export class PersonInputComponent {
+    constructor(private store: Store<any>) { }
+
+    add(name) {
+        this.store.dispatch({ type: 'ADD_PERSON', payload: {
+            name: name,
+        });
+    }
+}
+```
+
+---
+
+## `@Component` use `Actions` to communicate
+
+1. Responsible for Presentation and User Interractions
+1. It is blind to the complexities of the app (Doesn't know what happens after action is broadcasted)
+1. React to state change via select
+1. Affect the state via Actions
+1. Describes a need without dictating how to fulfill the need
+1. Eg. Need => MAIL_REQUIRED
+    1. `personOne` reacts by => `startWalking()`
+    1. `personTwo` reacts by => `callTaxi()`
+
+---
+
+## Store
+
+1. Responsible for application state (Immutability, Persisting to Disk etc.)
+1. The [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth)
+1. Keeper of stateful information, void of business logic
```

Comment:
Probably don't need the first part of this sentence since it's mentioned in the first point